serverSocket.js

```diff
@@ -0,0 +1,57 @@
+const io = require('socket.io')();
+const { UserChat } = require('./models/userChat');
+const { authMiddleware } = require('./middlewares/socket.middleware');
+
+// Initialize the socket.io server
+function init(server) {
+  io.attach(server);
+
+  async function getChats(userId) {
+    const [error, userChats] = await UserChat.find({ user: userId })
+      .populate('chat')
+      .exec();
```

Comment:
Populate the messages: https://mongoosejs.com/docs/populate.html#deep-populate
```suggestion
    const [error, userChats] = await UserChat.find({ user: userId })
      .populate({
        path: 'chat',
        populate: { path: 'messages' }
      })
      .exec();
```